src/services/heartbeat.test.ts

```diff
@@ -0,0 +1,92 @@
+import http from "node:http";
+import { expect, resolveOnEvent } from "../utils/tests-setup";
+import { configurableInitialize, interval } from "./heartbeat";
+import EventEmitter from "node:events";
+import { AddressInfo } from "node:net";
+import { Env } from "../utils/env";
+import { BotEventNames } from "../bot-events";
+import { waitFor } from "../bot";
+
+const createFakeServer = async (statusCode: number, onQuery?: () => void) => {
+  const server = http.createServer((_, res) => {
+    if (onQuery) onQuery();
+    res.writeHead(statusCode);
+    res.end();
+  });
+
+  await new Promise((resolve) => server.listen(0, () => resolve(null)));
+
+  return server;
+};
+
+describe("Heartbeat", () => {
+  const fn = configurableInitialize;
+
+  afterEach(() => {
+    if (interval) clearInterval(interval);
+  });
+
+  describe("initialize", () => {
+    it("should send a heartbeat every interval", async () => {
+      const expectedCalls = 4;
+
+      let httpCalls = 0;
+      const server = await createFakeServer(200, () => httpCalls++);
+      const { port } = server.address() as AddressInfo;
+
+      const emitter = new EventEmitter();
+      let sentEvents = 0;
+      emitter.on(BotEventNames.HEARTBEAT_SENT, () => sentEvents++);
+
+      const interval = 100;
+      const env = {
+        HEARTBEAT_URL: `http://localhost:${port}/test`,
+        HEARTBEAT_INTERVAL: interval,
+      } as Env;
+
+      const transportReadyEventPromise = resolveOnEvent(BotEventNames.HEARTBEAT_READY, emitter);
+
+      fn({ emitter, env });
+
+      await waitFor(expectedCalls * interval + interval * 0.5);
+      server.close();
+
+      expect(httpCalls).to.equal(expectedCalls, "web server received less calls than expected");
```

Comment:
(Ignorable)
Here could be less or more calls than expected, right? The error text should be "web server received a different amount of calls than expected" or something like that.
The same for the line below.

Reply:
@salgozino you are completely right, thanks for catching it! fixing it...

Reply:
Fixed! Thanks @salgozino